### DISCUSSION TITLE
What exactly is Branding?

### DISCUSSION THREAD

**Discussion by kamil7x** [forem/forem#17163] [Q&A] [answered]
On DEV, I can set two Brand colors.
By default Brand Color 1 is #000000 and Brand Color 2 is #ffffff.
My profile bg looks like this:
![image](https://user-images.githubusercontent.com/7629808/162214990-8534547a-0c93-455e-8041-2e098af3819a.png)

I tried to change both of them, and if I set color 1 to something else, eg. #be1a1a, then color 1 is used as my profile background:
![image](https://user-images.githubusercontent.com/7629808/162214480-2eca826a-e83c-46e1-9194-9402cae4fd4b.png)

Then if I set color 2 to eg. #1f61da then color 2 is used as background and I can't see any place where color 1 is used:
![image](https://user-images.githubusercontent.com/7629808/162214697-94f6aaf4-abc7-496f-bab0-0e6d2860d5e4.png)

More: if I set color 1 to #ffffff and color 2 to #000000 (default values inverted), result is same as with default values
![image](https://user-images.githubusercontent.com/7629808/162214933-76975f1b-94e3-4cce-82ca-b2f8835e3898.png)

So the question is: what is happening here?

**Comment by kamil7x** [comment]
Just to clarify, these are setups:
1.
![image](https://user-images.githubusercontent.com/7629808/162216205-62f4dcc8-0b70-467b-8882-acc44dc40631.png)

2.
![image](https://user-images.githubusercontent.com/7629808/162216297-29843e32-7677-4257-b2a6-552b391c88ea.png)

3.
![image](https://user-images.githubusercontent.com/7629808/162216374-d9f5bdc5-612f-41c8-96dd-8e191057915d.png)

4.
![image](https://user-images.githubusercontent.com/7629808/162216479-74056a91-d443-4cfb-93a1-4d034b20ffb2.png)

**Reply by kamil7x** [reply]
This one is interesting:
![image](https://user-images.githubusercontent.com/7629808/162216686-b6ce556d-1b90-4472-94bb-aea0f00ca20f.png)

Result:
![image](https://user-images.githubusercontent.com/7629808/162216730-d21984b5-669e-4f0e-925c-68ed25f004f6.png)

**Comment by kamil7x** [comment]
@JennieOcken @ludwiczakpawel sorry for bothering, but I really would like to understand how it works ;)

**Comment by ludwiczakpawel** [comment, marked as answer]
Even though we provide two fields for defining brand colors, we actually always use one: the one that has better contrast ratio (which we check with our weird internal algorithm...).

We used to use both colors to define more accents on the profile view, but we stopped doing so without removing the extra field. The only reason for NOT removing 2nd field at that time was literally lack of resources to do it. So it's now considered tech debt and/or legacy... And yes, we have to revisit that at some point - it's just been pretty low on the priority list so far. :(

**Reply by kamil7x** [reply]
Thank you very much for explanation

**Comment by amywtlin** [comment]
An issue has been created to address the confusion: https://github.com/forem/forem/issues/17686
